pkg/resource/logging-secret/alloy-logging-secret.go

```diff
@@ -0,0 +1,106 @@
+package loggingsecret
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"text/template"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/Masterminds/sprig/v3"
+
+	"github.com/giantswarm/logging-operator/pkg/common"
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	loggingcredentials "github.com/giantswarm/logging-operator/pkg/resource/logging-credentials"
+)
+
+//go:embed alloy/logging.alloy.template
+var alloyLogging string
+
+//go:embed alloy/logging-secret.yaml.template
+var alloyLoggingSecret string
+
+func GenerateAlloyLoggingSecret(lc loggedcluster.Interface, credentialsSecret *v1.Secret, lokiURL string) (v1.Secret, error) {
+	secret, err := generateAlloyLoggingConfig(lc, credentialsSecret, lokiURL)
+	if err != nil {
+		return v1.Secret{}, err
+	}
+
+	return secret, nil
+}
+
+func generateAlloyLoggingConfig(lc loggedcluster.Interface, credentialsSecret *v1.Secret, lokiURL string) (v1.Secret, error) {
+	var values bytes.Buffer
+
+	t, err := template.New("logging-config.yaml").Funcs(sprig.FuncMap()).Parse(alloyLoggingSecret)
```

Comment:
I think templates should be parsed in a constructor function to avoid doing it on every run because this is using some resources

Reply:
Moved this into the `init()` function

Reply:
I was thinking about a New function but that works :)